lib/cli/src/generators/generator.ts

```diff
@@ -0,0 +1,80 @@
+import { NpmOptions } from '../NpmOptions';
+import { StoryFormat, SupportedLanguage, SupportedFrameworks } from '../project_types';
+import {
+  retrievePackageJson,
+  getVersionedPackages,
+  writePackageJson,
+  getBabelDependencies,
+  installDependencies,
+  copyComponents,
+} from '../helpers';
+import configure from './configure';
+
+export type GeneratorOptions = {
+  language: SupportedLanguage;
+  storyFormat: StoryFormat;
+};
+
+export interface FrameworkOptions {
+  extraPackages?: string[];
+  extraAddons?: string[];
+  staticDir?: string;
+  addScripts?: boolean;
+  addComponents?: boolean;
+}
+
+export type Generator = (npmOptions: NpmOptions, options: GeneratorOptions) => Promise<void>;
+
+const defaultOptions: FrameworkOptions = {
+  extraPackages: [],
+  extraAddons: [],
+  staticDir: undefined,
+  addScripts: true,
+  addComponents: true,
+};
+const generator = async (
```

Comment:
Maybe it can be named `baseGenerator` and exported directly (I mean `export const baseGenerator =`)?